test/end-to-end/company.js

```diff
@@ -226,7 +226,571 @@ describe("Company endpoint", () => {
+                expect(res.body).toHaveProperty(
+                    "company._id",
+                    test_company_with_hidden_offer._id.toString()
+                );
+            });
+
+            test("should fail if company is disabled", async () => {
+                const res = await test_agent
+                    .get(`/company/${test_company_disabled._id}`)
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+
+                expect(res.body).toHaveProperty(
+                    "error_code",
+                    ErrorTypes.VALIDATION_ERROR
+                );
+                expect(res.body).toHaveProperty(
+                    "errors",
+                    expect.arrayContaining([
+                        expect.objectContaining({
+                            param: "companyId",
+                            msg: ValidationReasons.COMPANY_NOT_FOUND(
+                                test_company_disabled._id
+                            ),
+                        }),
+                    ])
+                );
+            });
+
+            test("should fail if company is blocked", async () => {
+                const res = await test_agent
+                    .get(`/company/${test_company_blocked._id}`)
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+
+                expect(res.body).toHaveProperty(
+                    "error_code",
+                    ErrorTypes.VALIDATION_ERROR
+                );
+                expect(res.body).toHaveProperty(
+                    "errors",
+                    expect.arrayContaining([
+                        expect.objectContaining({
+                            param: "companyId",
+                            msg: ValidationReasons.COMPANY_NOT_FOUND(
+                                test_company_blocked._id
+                            ),
+                        }),
+                    ])
+                );
+            });
+
+            test("should fail if company hasn't finished registration", async () => {
+                const res = await test_agent
+                    .get(
+                        `/company/${test_company_with_unfinished_registration._id}`
+                    )
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+
+                expect(res.body).toHaveProperty(
+                    "error_code",
+                    ErrorTypes.VALIDATION_ERROR
+                );
+                expect(res.body).toHaveProperty(
+                    "errors",
+                    expect.arrayContaining([
+                        expect.objectContaining({
+                            param: "companyId",
+                            msg: ValidationReasons.COMPANY_NOT_FOUND(
+                                test_company_with_unfinished_registration._id
+                            ),
+                        }),
+                    ])
+                );
+            });
+        });
```

Comment:
Bump on a comment made above, this should return a 404.

Also, the 'getAllCompanies' route returns companies that have not finished their registration, which is the opposite oh what this route is doing, what should be changed ?